models/repo.go

```diff
@@ -1515,7 +1515,6 @@ func DeleteRepository(doer *user_model.User, uid, repoID int64) error {
 		&Comment{RefRepoID: repoID},
 		&CommitStatus{RepoID: repoID},
 		&DeletedBranch{RepoID: repoID},
-		&webhook.HookTask{RepoID: repoID},
```

Comment:
Then we need to delete hook tasks via webhook's repo_id

Reply:
I see no problem with not deleting it here. I even think deleting it was harmful.

Currently we call `notification.NotifyDeleteRepository` before calling `models.DeleteRepository`. From there we can see three different posibilities:
1. The webhook gets send, the recipient reacts and is able to get informations about the repo by api (repo is not deleted at this moment)
2. The webhook gets send, the recipient reacts and is not able to get informations because the repo is already deleted
3. The webhook gets deleted and is not send at all (worst case for me)

With my change only case 1 and 2 are possible and the RepoDeleted webhook is always delivered.

Reply:
So maybe move `notification.NotifyDeleteRepository` after `models.DeleteRepository` is the right direction. If we don't delete the records even if repository deleted, we will keep many dirty data.

Reply:
If you create them after deleting the repository they will still be "dirty data". But I see now what you talk about. But we can simply delete them by using the `WebHook` struct as reference. If implemented, my version gets problem 3 too.

Reply:
oh :D